### DISCUSSION TITLE
Expand symbols with attributes at parse time

### DISCUSSION THREAD

**Discussion by peaceamongworlds** [hylang/hy#2001] [Ideas]
Is there a reason not to do this? It seems to me that it will make writing macros easier, especially those that substitute symbols within their body, like anaphoric macros or `let`.

**Comment by scauligi** [comment]
As in, have `x.y.z` at parse time become `(. x y z)` before macro expansion?

**Reply by peaceamongworlds** [reply]
yes

**Reply by scauligi** [reply]
I don't see why not, seems reasonable to me
